python/ql/lib/semmle/python/security/dataflow/UrlRedirectCustomizations.qll

```diff
@@ -59,6 +107,30 @@ module UrlRedirect {
     }
   }
 
+  /**
+   * A call to replace backslashes with forward slashes or eliminates them
+   * altogether, considered as a partial sanitizer, as well as an additional
+   * flow step.
+   */
+  class ReplaceBackslashesSanitizer extends Sanitizer, AdditionalFlowStep, DataFlow::MethodCallNode {
+    ReplaceBackslashesSanitizer() {
+      this.getFunction().(DataFlow::AttrRead).getAttributeName() = "replace" and
+      this.getArg(0).asExpr().(StrConst).getText() = "\\" and
+      this.getArg(1).asExpr().(StrConst).getText() in ["/", ""]
+    }
+
+    override predicate sanitizes(FlowState state) { state instanceof MayContainBackslashes }
+
+    override predicate step(
+      DataFlow::Node nodeFrom, FlowState stateFrom, DataFlow::Node nodeTo, FlowState stateTo
+    ) {
+      nodeFrom = this.getObject() and
```

Comment:
We usually discourage the separate use of `getObject` and `getMethodName` for `MethodCallNode`s, as these may not pair up uniquely (cf. the `calls` method). In this case, however, I think it's fairly unlikely for any problems to arise.

Reply:
Good point! I'll stick the receiver into a field to make sure it's paired up correctly with the method name in the charpred.